src/utilities/new-sequence-editor/to-seq-json.ts

```diff
@@ -251,60 +252,45 @@ export function parseTime(commandNode: SyntaxNode, text: string): Time {
     const timeTagEpochText = text.slice(timeTagEpochNode.from + 1, timeTagEpochNode.to).trim();
 
     // a regex to determine if this string [+/-]####T##:##:##.###
-    let match = testTime(timeTagEpochText, EPOCH_TIME);
-    if (match) {
-      const [, sign, doy, hh, mm, ss, ms] = match;
-      tag = `${sign === '-' ? '-' : ''}${doy !== undefined ? doy : ''}${hh ? hh : '00'}:${mm ? mm : '00'}:${
-        ss ? ss : '00'
-      }${ms ? ms : ''}`;
+    if (validateTime(timeTagEpochText, TimeTypes.EPOCH)) {
+      const { isNegative, days, hours, minutes, seconds, milliseconds } = parseDuration(timeTagEpochText);
+      tag = `${isNegative ? '-' : ''}${days !== 0 ? String(days).padStart(3, '0') : ''}${String(hours).padStart(2, '0')}:${String(minutes).padStart(2, '0')}:${String(
```

Comment:
This logic for adding padding to time strings seems to be reused in a few places. Is this something we can break out into its own function? It's also semi present in `time.ts`

Reply:
I can update the object to have a `time` entry which will store the above value. 

```
export type ParsedDurationString = {
  days: number;
  hours: number;
  isNegative: boolean;
  microseconds: number;
  milliseconds: number;
  minutes: number;
  seconds: number;
  years: number;
};
```

I think a padding function is overkill as it doesn't save much.

String(s).padStart(n,'0') -> addPadding(s,n)

Reply:
Sorry, I don't mean just a padding function, I mean a function that converts every part of the `ParsedDurationString` into an appropriate object with all the padding added to every part. It seems like the overall combined string output can be different for different time types, but the padding for each part seems consistent.

Reply:
I made another function called `getDurationTimeComponents`